### ISSUE TITLE
distupgrade all 

### ISSUE THREAD

**Issue by mluscon** [openSUSE/libsolv#124]
Hi,

we found another issue in 0.6.19 regarding distro upgrade. 

Following distupgrade:
```
repo system 0 testtags <inline>
#>=Pkg: hole 1 1 x86_64
#>=Pkg: librita 1 1 i686
#>=Pkg: librita 1 1 x86_64
#>=Pkg: pepper 20 0 x86_64
#>=Req: librita >= 1-0
#>=Prv: parking
#>=Prv: /raised/smile
#>=Pkg: tour 5 0 noarch
#>=Pkg: dup 1 0 noarch
#>=Pkg: dup 2 0 noarch
#>=Pkg: dup 3 0 noarch
repo available 0 testtags <inline>
#>=Pkg: pepper 19 0 x86_64
#>=Req: librita >= 1-0
#>=Prv: /raised/smile
#>=Pkg: librita 1.1 1 x86_64
#>=Pkg: librita 1.1 1 i686
#>=Pkg: tour 5 0 noarch
system x86_64 * system
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
poolflags implicitobsoleteusescolors
job distupgrade all packages
result transaction,problems <inline>
```

returns:
```
+downgrade pepper-20-0.x86_64@system pepper-19-0.x86_64@available
+problem a28a55b2 info problem with installed package dup-1-0.noarch
+problem a28a55b2 solution 2036c5d0 erase dup-1-0.noarch@system
+problem a28a55b2 solution 2036c5d0 erase dup-2-0.noarch@system
+problem a28a55b2 solution 2036c5d0 erase dup-3-0.noarch@system
+problem a28a55b2 solution f5533b23 allow dup-1-0.noarch@system
+upgrade librita-1-1.i686@system librita-1.1-1.i686@available
+upgrade librita-1-1.x86_64@system librita-1.1-1.x86_64@available
```
and expected is just:
```
+downgrade pepper-20-0.x86_64@system pepper-19-0.x86_64@available
+upgrade librita-1-1.i686@system librita-1.1-1.i686@available
+upgrade librita-1-1.x86_64@system librita-1.1-1.x86_64@available
```

**Comment by mlschroe** [comment]
Problem in the orphan detection. It's not considered an orphan because there are different versions installed. I'll fix this tomorrow.

**Comment by mluscon** [comment]
Thanks a lot.

**Comment by mlschroe** [comment]
Hopefully fixed with commit 6c9cef4

**Comment by mluscon** [comment]
Yes, it is fixed now. Thank you!
